libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -264,28 +269,33 @@ bool Compass::send_mag_cal_progress(const GCS_MAVLINK& link)
 
 bool Compass::send_mag_cal_report(const GCS_MAVLINK& link)
 {
-    uint8_t cal_mask = _get_cal_mask();
+    const mavlink_channel_t chan = link.get_chan();
+
+    for (uint8_t i = 0; i < COMPASS_MAX_INSTANCES; i++) {
+        Priority compass_id = (next_cal_report_idx[chan] + 1) % COMPASS_MAX_INSTANCES;
 
-    for (Priority compass_id(0); compass_id<COMPASS_MAX_INSTANCES; compass_id++) {
-        // ensure we don't try to send with no space available
-        if (!HAVE_PAYLOAD_SPACE(link.get_chan(), MAG_CAL_REPORT)) {
-            // ideally we would only send one progress message per
-            // call.  If we don't return true here we may end up
-            // hogging *all* the bandwidth
-            return true;
-        }
         if (_calibrator[compass_id] == nullptr) {
+            next_cal_report_idx[chan] = compass_id;
             continue;
         }
         const CompassCalibrator::Report cal_report = _calibrator[compass_id]->get_report();
         if (cal_report.status == CompassCalibrator::Status::SUCCESS ||
```

Comment:
Doing early-return here would avoid having to do the assignment into next_cal_report_idx[] twice.  Doesn't need to be done as part of this PR, of course - but OTOH most of the indented block is being replaced anyway!

Reply:
Well, this was wrong; stylistically I prefer the early-continue, but it doesn't save that assignment.

Reply:
I'm going to leave this as boils down to a style change, and isn't terser.